config_manager.py

```
@@ -0,0 +1,348 @@
+                backup_path = self.config_yml_path.parent / f"config.yml.backup.{timestamp}"
+                shutil.copy2(self.config_yml_path, backup_path)
+                logger.info(f"Backed up config.yml to {backup_path.name}")
+
+            # Write updated config
+            with open(self.config_yml_path, 'w') as f:
+                yaml.dump(config, f, default_flow_style=False, sort_keys=False)
+
+            logger.info(f"Saved config.yml to {self.config_yml_path}")
+
+        except Exception as e:
+            logger.error(f"Failed to save config.yml: {e}")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)